File: app/schemas/payment.py
# File: app/schemas/payment.py

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.models.enums import PaymentMethod, PaymentStatus


class PaymentSubmitRequest(BaseModel):
    """بيانات رفع إثبات دفع (بنكك/فيزا) من العميل أو الوكيل."""

    payment_method: PaymentMethod
    amount: Decimal
    currency_code: str | None = None
    transaction_ref: str | None = None


class PaymentVerifyRequest(BaseModel):
    """قرار موظف/مدير بعد مراجعة إثبات الدفع: قبول أو رفض."""

    approve: bool
    notes: str | None = None


class PaymentOut(BaseModel):
    """تمثيل سجل دفع كامل في الاستجابات، مع رابط موقّت لصورة إشعار الدفع."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    order_id: int
    order_number: str
    payment_method: PaymentMethod
    amount: Decimal
    currency_code: str | None
    transaction_ref: str | None
    receipt_signed_url: str | None
    status: PaymentStatus
    verified_by: int | None
    verified_at: datetime | None
    created_at: datetime

File: frontend/src/types/payment.ts
// File: frontend/src/types/payment.ts

import type { PaymentMethod, PaymentStatus } from "@/types/enums";

/** مطابق لـ app.schemas.payment.PaymentOut. */
export interface PaymentOut {
  id: number;
  order_id: number;
  order_number: string;
  order_total_amount: string;
  order_currency_code: string;
  payment_method: PaymentMethod;
  amount: string;
  currency_code: string | null;
  transaction_ref: string | null;
  receipt_signed_url: string | null;
  status: PaymentStatus;
  verified_by: number | null;
  verified_at: string | null;
  created_at: string;
}

/** مطابق لـ app.schemas.payment.PaymentVerifyRequest. */
export interface PaymentVerifyRequest {
  approve: boolean;
  notes: string | null;
}
